test/session_spec.lua

```diff
@@ -168,6 +168,18 @@ test_session("Session using ChidProcessStream", function ()
   return Session.new(proc_stream)
 end)
 
+describe("ChildProcessStream", function()
+  it("can capture exit code", function()
+    local proc_stream = ChildProcessStream.spawn({
+      nvim_prog, '-u', 'NONE', '--embed',
+    })
+
+    proc_stream:close()
+
+    assert.is_not_nil(proc_stream._exit_code)
```

Comment:
why not `assert(0 == proc_stream._exit_code)` ?

Reply:
Because I wanted to only capture the fact that the exit code was set, and not necessarily check it's value.

Reply:
But more precision is good, if possible. The test controls the `nvim` invocation, so why not test that the exit code is actually a correct result?

Reply:
Let's do that then